packages/@lwc/integration-karma/test/custom-elements/index.spec.js

```diff
@@ -163,49 +203,62 @@ if (SUPPORTS_CUSTOM_ELEMENTS) {
-            expect(secondCtor).not.toBeUndefined();
-            expect(secondCtor).toBe(firstCtor);
-        });
+        describe('same constructor returned from get/whenDefined', () => {
+            function defineFirstElement(tagName, Component) {
+                // For the scoped registry, we want to test the global customElements API and
+                // make sure it consistently returns the same constructor. For vanilla custom elements,
+                // we can test lwc.createElement instead since it's also registered globally.
+                if (window.lwcRuntimeFlags.ENABLE_SCOPED_CUSTOM_ELEMENT_REGISTRY) {
+                    customElements.define(tagName, Component.CustomElementConstructor);
+                } else {
+                    createElement(tagName, { is: Component });
+                }
+            }
 
-        it('whenDefined() should always return the same constructor - defined before whenDefined', () => {
-            createElement('x-nonce6', { is: Nonce6 });
-            let firstCtor;
-            return customElements
-                .whenDefined('x-nonce6')
-                .then((Ctor) => {
-                    expect(Ctor).not.toBeUndefined();
-                    firstCtor = Ctor;
-                    createElement('x-nonce6', { is: Nonce7 });
-                    return customElements.whenDefined('x-nonce6');
-                })
-                .then((Ctor) => {
-                    expect(Ctor).not.toBeUndefined();
-                    expect(Ctor).toBe(firstCtor);
-                });
-        });
+            it('get() should always return the same constructor', () => {
+                defineFirstElement('x-nonce10', Nonce10);
+                const firstCtor = customElements.get('x-nonce10');
+                expect(firstCtor).not.toBeUndefined();
+                createElement('x-nonce10', { is: Nonce11 });
+                const secondCtor = customElements.get('x-nonce10');
+                expect(secondCtor).not.toBeUndefined();
+                expect(secondCtor).toBe(firstCtor);
+            });
 
-        it('whenDefined() should always return the same constructor - defined after whenDefined', () => {
-            const promise = customElements.whenDefined('x-nonce12');
-            createElement('x-nonce12', { is: Nonce12 });
-            let firstCtor;
-            return promise
-                .then((Ctor) => {
-                    expect(Ctor).not.toBeUndefined();
-                    firstCtor = Ctor;
-                    const promise = customElements.whenDefined('x-nonce12');
-                    createElement('x-nonce12', { is: Nonce13 });
-                    return promise;
-                })
-                .then((Ctor) => {
-                    expect(Ctor).not.toBeUndefined();
-                    expect(Ctor).toBe(firstCtor);
-                });
+            it('whenDefined() should always return the same constructor - defined before whenDefined', () => {
```

Comment:
```suggestion
            it('whenDefined() should always return the same constructor - pivot defined before whenDefined', () => {
```

Reply:
This test actually runs with and without pivots.